ssl/ssl_cert.c

```diff
@@ -68,23 +68,18 @@ CERT *ssl_cert_new(size_t ssl_pkey_num)
     if (!ossl_assert(ssl_pkey_num >= SSL_PKEY_NUM))
         return NULL;
 
-    ret = OPENSSL_zalloc(sizeof(*ret));
+    ret = OPENSSL_zalloc(sizeof(*ret) + ssl_pkey_num * sizeof(CERT_PKEY));
```

Comment:
You cannot do this unfortunately. What you would have to do with non-standard GCC extension is to add 0-sized array of CERT_PKEY at the end of the CERT struct, however that is GCC-only construct. With C99 you would use `CERT_PKEY cpkey[];` as the last member of CERT. Unfortunately we do not allow C99.

So please revert this allocation change.